ingest/mongo_connection.py

```diff
@@ -83,19 +83,34 @@ def wrapper(*args, **kwargs):
                         "Batch ops error occurred. Reinsert attempt %s.", str(attempt)
                     )
                     error_docs = bwe.details["writeErrors"]
-                    # Check error code to see if any failures are due to duplicate ids (error code 11000)
-                    if any(doc["code"] == 11000 for doc in error_docs):
-                        args[0] = reassign_id(args[0])
-                    retry(attempt)
+                    # Check error code to see if any failures are due to violating a unique index (error code 11000)
+                    # and discard those documents before retrying
+                    filtered_docs = discard_inserted_documents(error_docs, args[0])
+                    if len(filtered_docs) > 0:
+                        args[0] = filtered_docs
+                        retry(attempt)
+                    else:
+                        return args[0]
                 else:
                     dev_logger.debug(str(bwe.details))
```

Comment:
There won't be any file attributes available - the only thing we'll have here is the PyMongo operation and the raw documents.

Reply:
I'm going to open a ticket for this, because the `MetricsService` class has some weirdness for setting properties on events.  The proposed solution errors out, and I don't want to hold this up for that, but I agree it should get done.

Reply:
https://broadworkbench.atlassian.net/browse/SCP-3714